website/static/js/share/utils.js

```diff
@@ -26,57 +30,124 @@ var formatNumber = function(num) {
 
 var loadingIcon = m('img[src=/static/img/loading.gif]',{style: {margin: 'auto', display: 'block'}});
 
-var loadMore = function(vm) {
-    if (vm.query().length === 0) {
+utils.errorState = function(vm){
+    vm.results = null;
+    vm.statsData = undefined;
+    vm.time = 0;
+    vm.count = 0;
+    vm.resultsLoading(false);
+    m.redraw(true);
+    $osf.growl('Error', 'invalid query');
+};
+
+utils.updateVM = function(vm, data) {
+    if (data === null) {
         return;
     }
-    var page = vm.page++ * 10;
-    vm.resultsLoading(true);
-
-    m.request({
-        method: 'get',
-        background: true,
-        url: '/api/v1/share/?' + $.param({
-            from: page,
-            q: vm.query(),
-            sort: 'dateUpdated',
-        })
-    }).then(function(data) {
-        vm.time = data.time;
-        vm.count = data.count;
-        data.results.forEach(function(result) {
-            result.title = scrubHTML(result.title);
-            result.description = scrubHTML(result.description);
-        });
-
-        vm.results.push.apply(vm.results, data.results);
-
-        vm.resultsLoading(false);
-    }).then(m.redraw).then(function() {
-        callbacks.map(function(cb) {cb();});
+    vm.time = data.time;
+    vm.count = data.count;
+    data.results.forEach(function(result) {
+        result.title = utils.scrubHTML(result.title);
+        result.description = utils.scrubHTML(result.description);
+    });    
+    vm.results.push.apply(vm.results, data.results);   
+    m.redraw();
+    $.map(callbacks, function(cb) {
+        cb();
     });
 };
 
-var search = function(vm) {
-    if (vm.query().length === 0) {
-        return;
+utils.loadMore = function(vm) {    
```

Comment:
Hm, if the `ViewModel` is the bundle of state shared among the ShareApp's controllers, then the `ViewModel`'s methods would be the behaviors for modifying that shared state. The methods provide the interface by which controllers interact with the shared state.

That makes sense to me more than utils functions that can only ever be called with `ShareApp.ViewModel` as their argument. 

But hey, we're treading into bikeshedding territory. Follow your :heart: